src/App.tsx

```diff
@@ -1,14 +1,69 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+function checkQuery(string: string, query: string): boolean {
+  const preparedString = string.toLowerCase().trim();
+  const preparedQuery = query.toLowerCase().trim();
+
+  return preparedString.includes(preparedQuery);
+}
+
+function filterTodos(todos: Todo[], query: string, filterField: FilterType) {
+  let filteredTodos = [...todos];
```

Comment:
no need to copy here

```suggestion
  let filteredTodos = todos;
```